### DISCUSSION TITLE
rosbrige_server on HSR gazebo simulator and HSR hardware

### DISCUSSION THREAD

**Discussion by jjnak** [HumanSupportRobot/QandA#79] [Q&A] [answered]
I have a question on rosbridge_server on HSR gazebo simulator and HSR hardware.

When I tried the following
on Terminal 1
`$ roslaunch hsrb_gazebo_launch hsrc_empty_world.launch `

on Terminal 2
`$ roslaunch rosbridge_server rosbridge_websocket.launch`

I had the following warning:
`2023-09-01 15:58:35+0900 [-] [WARN] [1693551515.838461, 0.000000]: Unable to start server: Couldn't listen on any:9090: [Errno 98] Address already in use. Retrying in 5s.`

This means that it looks like rosbridge_server is already running on HSR gazebo simulator since I was able to pub/sub messages using rosbridge protocol without separately running rosbridge_server (only after running HSR gazebo simulator). I'd like to know if rosbridge_server is running on HSR gazebo simulator and also on HSR hardware by default.

Thanks,

-Jun

**Comment by k-takeshita** [comment, marked as answer]
Yes, rosbridge_server is running by default on both real robot and in the simulator.

**Reply by jjnak** [reply]
Thank you for your response!
